common/event.c

```diff
@@ -579,6 +582,14 @@ event_inotify_fd(void)
 	event_io_t *io = event_io_new(fd, EVENT_IO_READ, &event_inotify_cb, NULL);
 	event_add_io(io);
 
+	if (io) {
+		event_inotify_io = io;
+		TRACE("Setting event_inotify_io=%p (fd=%d)", (void *)event_inotify_io,
+		      event_inotify_io->fd);
+	} else {
+		FATAL_ERRNO("Could not init event_inotify_io (fd=%d)", fd);
+	}
+
```

Comment:
the "if (io)"-block should be moved above event_add_io(io);
Does not make sence to call add_io if io == NULL;